docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,841 @@
+| description | string                                                                                            | 1-100 character description    |
+| options?    | array of [ApplicationCommandOption](#DOCS_INTERACTIONS_SLASH_COMMANDS/applicationcommandoption) | the parameters for the command |
+
+## Edit Guild Application Command % PATCH /applications/{application.id#DOCS_TOPICS_OAUTH2/application-object}/guilds/{guild.id#DOCS_RESOURCES_GUILD/guild-object}/commands/{command.id#DOCS_INTERACTIONS_SLASH_COMMANDS/applicationcommand}
+
+Edit a guild command. Updates for guild commands will be available immediately. Returns `200` and an [ApplicationCommand](#DOCS_INTERACTIONS_SLASH_COMMANDS/applicationcommand) object.
+
+###### JSON Params
+
+| Field       | Type                                                                                              | Description                    |
+|-------------|---------------------------------------------------------------------------------------------------|--------------------------------|
+| name        | string                                                                                            | 3-32 character command name    |
+| description | string                                                                                            | 1-100 character description    |
+| options?    | array of [ApplicationCommandOption](#DOCS_INTERACTIONS_SLASH_COMMANDS/applicationcommandoption) | the parameters for the command |
+
+
+## Delete Guild Application Command % DELETE /applications/{application.id#DOCS_TOPICS_OAUTH2/application-object}/guilds/{guild.id#DOCS_RESOURCES_GUILD/guild-object}/commands/{command.id#DOCS_INTERACTIONS_SLASH_COMMANDS/applicationcommand}
+
+Delete a guild command. Returns `204` on success.
+
+## Create Interaction Response % POST /interactions/{interaction.id#DOCS_INTERACTIONS_SLASH_COMMANDS/interaction}/{interaction.token#DOCS_INTERACTIONS_SLASH_COMMANDS/interaction}/callback
+
+Create a response to an Interaction from the gateway. Takes an [Interaction response](#DOCS_INTERACTIONS_SLASH_COMMANDS/interaction-response).
+
+## Edit Original Interaction Response % PATCH /webhooks/application.id/{interaction.token#DOCS_INTERACTIONS_SLASH_COMMANDS/interaction}/messages/@original
+
+Edits the initial Interaction response. Functions the same as [Edit Webhook Message](#DOCS_RESOURCES_WEBHOOK/edit-webhook-message).
+
+## Delete Original Interaction Response % DELETE /webhooks/application.id/{interaction.token#DOCS_INTERACTIONS_SLASH_COMMANDS/interaction}/messages/@original
+
+Deletes the initial Interaction response. Returns `204` on success.
+
+## Create Followup Message % POST /webhooks/application.id/{interaction.token#DOCS_INTERACTIONS_SLASH_COMMANDS/interaction}
+
+Create a followup message for an Interaction. Functions the same as [Execute Webhook](#DOCS_RESOURCES_WEBHOOK/execute-webhook)
```

Comment:
Can this use different interaction types (like `ChannelMessageWithSource`) or is it completely restricted to the parameters from webhooks? If it is restricted, will we be able to use `file` in interaction responses or would it only be here?

Reply:
Followups are only webhook messages, types are only for the initial response

Reply:
Advaith is correct that only the original response supports the different types since any followup message is just like sending a normal message.

I don't believe we support `file` yet, but we should. Had to make some tradeoffs in the architecture to start but definitely noted :)